src/binary/non_blocking/raw_binary_reader.rs

```diff
@@ -769,8 +770,8 @@ impl<A: AsRef<[u8]>> IonReader for RawBinaryBufferReader<A> {
         self.read_symbol_id().map(RawSymbolToken::SymbolId)
     }
 
-    fn read_blob(&mut self) -> IonResult<Vec<u8>> {
-        self.read_blob_bytes().map(Vec::from)
+    fn read_blob(&mut self) -> IonResult<Blob> {
+        self.read_blob_bytes().map(Vec::from).map(Blob::from)
```

Comment:
🗺️ The `IonReader` trait's blob- and clob- related methods had to be updated to return `Blob` and `Clob`.

Before:
```rust
fn read_blob(&mut self) -> IonResult<Vec<u8>>;
fn read_clob(&mut self) -> IonResult<Vec<u8>>;
```

After:
```rust
fn read_blob(&mut self) -> IonResult<Blob>;
fn read_clob(&mut self) -> IonResult<Clob>;
```

This change had to be made in each implementation of `IonReader`, of which there are several.